modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -1564,21 +1564,22 @@ public CommerceOrder updatePaymentStatusAndTransactionId(
 			String transactionId)
 		throws PortalException {
 
-		CommerceOrder commerceOrder = commerceOrderPersistence.findByPrimaryKey(
-			commerceOrderId);
+		CommerceOrder oldCommerceOrder =
+			commerceOrderPersistence.findByPrimaryKey(commerceOrderId);
 
-		int previousPaymentStatus = commerceOrder.getPaymentStatus();
+		CommerceOrder newCommerceOrder =
```

Comment:
I don't think this is the right solution:

> A shallow copy just copies the object reference into the target reference. It does not create a new object on the heap. By default, Java does shallow cloning using clone() function.

Reply:
I think you should use `cloneWithOriginalValues()`.
Also, I would keep the order that you modify "newOrder" as the one coming from the peristence and use the copy as "oldOrder" just to be consistent, otherwise you are switching the object retrieved and the one stored.